lgc/include/lgc/state/PalMetadata.h

```diff
@@ -0,0 +1,64 @@
+/*
+ ***********************************************************************************************************************
+ *
+ *  Copyright (c) 2020 Advanced Micro Devices, Inc. All Rights Reserved.
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in all
+ *  copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ *
+ **********************************************************************************************************************/
+/**
+ ***********************************************************************************************************************
+ * @file  PalMetadata.h
```

Comment:
Do you think about merging AbiMetadata.h with PalMetadata.h?
We still have Abi.h. Clearlyl, all those definitions are finally consumed by PAL. It seems the files are somewhat unnecessarily separated.

Reply:
I think there is value in having the three separate files:
1. PalMetadata.h/.cpp is specifically a class used by other code to set PAL metadata
2. AbiMetadata.h contains definitions relating to PAL metadata
3. Abi.h contains definitions relating to the PAL ABI, other than metadata.
Maybe you could argue that 2 and 3 could be merged. But PalMetadata.h should definitely remain separate, as it is a class declaration.

Reply:
It's OK to keep three files. What I worry about is if someone wants to add something new of ABI, he might put the stuffs in wrong place. Without your explanation, I even don' know the files serve such functions. Better to describe these in the file header comments.

Reply:
I added some comments.